app/controllers/chat_controller.rb

```diff
@@ -149,6 +149,14 @@ def update_user_last_read
 
     membership = UserChatChannelMembership.find_by(user: current_user, chat_channel: @chat_channel, following: true)
     raise Discourse::NotFound if membership.nil?
+
+    if membership.last_read_message_id && params[:message_id].to_i < membership.last_read_message_id
+      raise Discourse::InvalidParameters.new(:message_id)
+    end
+
+    chat_message = ChatMessage.find_by(chat_channel_id: @chat_channel.id, id: params[:message_id])
```

Comment:
Do we use `chat_message` anywhere else? If not, should we instead do an existence check rather than loading the model from the database?

Reply:
No indeed we don't, this should be enough: https://github.com/discourse/discourse-chat/pull/891/commits/693b54a034188860c88b72b8c15502a2c6f9b086